usaspending_api/awards/management/commands/loadfinancialassistance.py

```diff
@@ -0,0 +1,189 @@
+
+
+class Command(BaseCommand):
+    help = "Loads awards from a usaspending financial assistance download. \
+            Usage: `python manage.py loadcontracts source_file_path`"
+
+    logger = logging.getLogger('console')
+
+    def add_arguments(self, parser):
+        parser.add_argument('file', nargs=1, help='the file to load')
+
+    def handle(self, *args, **options):
+        # Create a new submission attributes object for this timestamp
+
+        subattr = SubmissionAttributes()
+        subattr.usaspending_update = datetime.now()
+        subattr.save()
+
+        # field_map is used to simply translate a column name in our schema
+        # to the field name in the input source
+        field_map = {
+            "award_id": "federal_award_id",
+            "federal_action_obligation": "fed_funding_amount",
+            "non_federal_funding_amount": "non_fed_funding_amount",
+            "cfda_number": "cfda_program_num",
+            "cfda_title": "cfda_program_title",
+            "uri": "unique_transaction_id",
+            "face_value_loan_guarantee": "face_loan_guran",
+            "original_loan_subsidy_cost": "orig_sub_guran",  # ??
+            "award_description": "project_description",  # ??
+        }
+
+        # TODO: csv contains `exec1_amount`... `exec5_amount` and
+        # `exec1_fullname`... `exec5_fullname`.  We seem to be ignoring those.
+
+        # TODO: What are all the drv_ fields in financial_assistance_award?
+
+        # progsrc - program source?
+        # I’ve also got a “progsrc_acnt_code”, “progsrc_agen_code”, “progsrc_subacnt_code”, which I’m guessing stands for Program Source, and corresponds to… Funding Agency as opposed to Awarding Agency?
+        # none of those really look like enough to identify an agency
+        # how to identify the funding agency?
+
+        value_map = {
+            "data_source": "USA",
+            "submitted_type": "C",  # ?? For CSV?
+            "recipient": self.get_or_create_recipient,
+            "award": self.get_or_create_award,
+            "place_of_performance":
+            lambda row: h.get_or_create_location(row, location_mapper_fin_assistance_principal_place),
+            "awarding_agency": self.get_awarding_agency,
+            "assistance_type": lambda row: h.up2colon(row['assistance_type']),
+            "record_type": lambda row: int(h.up2colon(row['record_type'])),
+            "action_type": lambda row: h.up2colon(row['action_type']),
+            "action_date":
+            lambda row: h.convert_date(row['obligation_action_date']),
+            "last_modified_date":
+            lambda row: h.convert_date(row['last_modified_date']),
+            "transaction_number":
+            lambda row: self.parse_first_character(row['transactionnumber']),
+            "solicitation_identifier":
+            lambda row: self.parse_first_character(row['solicitationid']),
+            "submission": subattr
+        }
+
+        loader = ThreadedDataLoader(
```

Comment:
A good addition here is to set `collision_field="usaspending_unique_transaction_id", collision_behavior="skip"` when you add it to the model, this way you don't reload data that is already there (unless you specifically want to!), in which case you can use `collision_behavior="delete"`